inflate.c

```diff
@@ -371,9 +371,18 @@ static int updatewindow(z_stream *strm, const unsigned char *end, uint32_t copy)
 
     /* if it hasn't been done already, allocate space for the window */
     if (state->window == NULL) {
+#ifdef INFFAST_CHUNKSIZE
+        unsigned wsize = 1U << state->wbits;
+        state->window = (unsigned char FAR *)
+                        ZALLOC(strm, wsize + INFFAST_CHUNKSIZE,
+                               sizeof(unsigned char));
```

Comment:
This is an interesting change that guarantees chunk memset and chunk memcpy operations to not run over the end of the allocated memory.  This has the benefit of removing some performance costly branches from the hot paths of the memory operations.